### ISSUE TITLE
Help with ISIC 2017 Dataset Issue

### ISSUE THREAD

**Issue by Sajjad-rafiee** [wurenkai/UltraLight-VM-UNet#65]
Hello,

I have downloaded the ISIC 2017 dataset files (Training Data, Test Data, and Validation Data) and performed preprocessing. However, when I run the model for training, the results are quite different from what I expected. Here are the results:

miou: 0.6619

f1_or_dsc: 0.7966

accuracy: 0.9372

specificity: 0.9854

sensitivity: 0.7079

confusion_matrix: [[8004527 118527]

Could you please guide me on where I might have gone wrong? I suspect the issue may be with the preprocessing step. Specifically, I added this line to the code:

img2 = np.round(img2).astype(np.float32)

I'd appreciate any suggestions or insights you might have.

Thank you in advance for your help!

**Comment by wurenkai** [comment]
Hi, if you add the “np.round” operation to the mask processing, would you consider adding the binarization operation before? Because during the “bilinear” operation, the size scaling causes pixel transitions to appear. I wonder if that is the cause?

**Comment by wurenkai** [comment]
Hi, that accuracy is definitely not normal, although there are some fluctuations with each different device, a DSC value of 0.79 is not normal. I'm going by your description that you have an anomaly in integrating our code into your project, so have you run it in our code project? As described in the paper, the model is also augmented with data through the configuration file during training, as you can see in the implementation details section of our paper.
